applicationset/services/scm_provider/utils.go

```diff
@@ -165,3 +168,11 @@ func getApplicableFilters(filters []*Filter) map[FilterType][]*Filter {
 	}
 	return filterMap
 }
+
+func checkBranchDoNotExist(error string) bool {
+	error = strings.ToLower(error)
+	if strings.Contains(error, "not found") || strings.Contains(error, "notfound") || strings.Contains(error, "404") || strings.Contains(error, "do not exists") {
```

Comment:
Can an error type be defined and returned by `provider.GetBranches(ctx, repo)` and avoid all these string comparisons? 

In this case we could do something like:
```go
branches, err := provider.GetBranches(ctx, repo)
if errors.Is(err, BranchNotFoundErr) {
 ...
}
```

Reply:
I guess it would be hard or impossible to do so GetBranches is interface that we define and its implementation is with Git provider eg: github,bitbucket or gitea, may be i can collect actual error message for all service provider and assign and arrange under BranchNotFoundErr it will give cleaner code

Reply:
The implementation is also provided by the applicationset codebase in the `scm_provider` package. What I am suggesting is that every implementation should check how the underlying scm provider returns the not found error and return a unified error type that can be used by the `SCMProviderService` consumers.

Reply:
Yeah could be done.

Reply:
Done

Reply:
Thank you. Do you plan to do it for the other scm providers as well?